packages/api/tests/mock.factories.js

```diff
@@ -1466,25 +1466,44 @@ function fakeDisease(defaultData = {}) {
   };
 }
 
+const volumeUnits = ['l', 'gal', 'ml', 'fl-oz'];
+const weightUnits = ['g', 'lb', 'kg', 't', 'mt', 'oz'];
+
 async function pest_control_taskFactory(
   { promisedTask = taskFactory(), promisedProduct = productFactory() } = {},
   pestTask = fakePestControlTask(),
 ) {
   const [task, product] = await Promise.all([promisedTask, promisedProduct]);
   const [{ task_id }] = task;
   const [{ product_id }] = product;
+
+  // The model normally handles this conversion, but this is a direct insert into the database
+  const isVolume = volumeUnits.includes(pestTask.product_quantity_unit);
+  const isWeight = weightUnits.includes(pestTask.product_quantity_unit);
+
+  const updatedPestTask = {
+    ...pestTask,
+    volume: isVolume ? pestTask.product_quantity : null,
+    volume_unit: isVolume ? pestTask.product_quantity_unit : null,
+    weight: isWeight ? pestTask.product_quantity : null,
```

Comment:
@Duncan-Brain I realized updating this test (yay, testing!) that the stubbed code needs logic to `null` the other property (weight/volume) or else the PATCH can violate the volume or weight check if `product_quantity_unit` switches types. Could you update the stub?

Reply:
Should be done!